src/rtsp_player_controller.cc

```diff
@@ -412,32 +416,35 @@ void RTSPPlayerController::RTPCheckAndSendBackStats(RTPDemuxContext *s, uint64_t
 }
 
 void RTSPPlayerController::StartParsing(int32_t) {
-	//Open the Audio Decoder Context
 	AVCodecContext *in_codec_ctx = NULL, *out_codec_ctx = NULL;
 	SwrContext *resample_context = NULL;
 	AVAudioFifo *fifo = NULL;
-	AVStream* s = format_context_->streams[audio_stream_idx_];
+	AVStream* s = NULL;
 	int ret = 0, bits_this_sec = 0;
 	uint64_t stats_last_sent = 0;
 	RTSPState *state;
 	RTPDemuxContext *demux;
 
-	init_transcoder(s->codecpar, &in_codec_ctx, &out_codec_ctx, &resample_context);
+	// Open the Audio Decoder Context
+	if (audio_stream_idx_ >= 0) {
+		s = format_context_->streams[audio_stream_idx_];
+		init_transcoder(s->codecpar, &in_codec_ctx, &out_codec_ctx, &resample_context);
+
+		// Initialize the FIFO buffer to store audio samples to be encoded
+		init_fifo(&fifo, out_codec_ctx);
 
-	// Initialize the FIFO buffer to store audio samples to be encoded
-	init_fifo(&fifo, out_codec_ctx);
+		// Audio Level update
+		prev_audio_ts_ = 0;
+		audio_level_ = 0;
+		is_parsing_finished_ = false;
```

Comment:
This flag is common for both audio and video, hence can be moved out of this if case.